spec/services/cycle_timetable_spec.rb

```diff
@@ -29,6 +29,16 @@
         expect(described_class.current_year).to eq(next_year)
       end
     end
+
+    it 'returns this_year for the date of `apply_opens`' do
+      travel_temporarily_to(described_class.apply_opens(this_year)) do
+        expect(described_class.current_year).to eq(this_year)
+      end
+    end
+
+    it 'returns 2022 for current_year(CycleTimetable.find_opens(2023))' do
+      expect(described_class.current_year(described_class.find_opens(2023))).to eq(2022)
+    end
```

Comment:
This may be surprising to understand. I tried to make changes to `find_opens` but discovered that a change to this portion of the cycle would require changes to the other portions of the cycle and that is too much effort to do here.

This test is to demonstrated the specification in the code, it may or may not be how we want or expect the portions of the cycle to work but demonstrates how the cycle portions actually behave.